File: test/binaries/15_objective_c/hello_world.m
/*
 * gcc -lobjc -framework Foundation hello_world.m
 */

#import <Foundation/Foundation.h>

@interface HelloWorld : NSObject
- (void)hello_world;
@end

@implementation HelloWorld
- (void)hello_world {
    NSString *message = @"Hello, World!";
    fputs(message.UTF8String, stdout);
    fputc('\n', stdout);
}
@end

int main(int argc, char *argv[])
{
    @autoreleasepool {
        HelloWorld *hw = [HelloWorld new];
        [hw hello_world];
    }
    return 0;
}

File: test/binaries/16_swift/hello_world.swift
/*
 * swiftc hello_world.swift -o hello_world
 */

import Foundation

final class HelloWorld {
    func hello_world() {
        let message = "Hello, World!"
        FileHandle.standardOutput.write(
            message.appending("\n").data(using: .utf8)!
        )
    }
}

autoreleasepool {
    let hw = HelloWorld()
    hw.hello_world()
}